packages/contracts/test/framework/dao/dao-factory.ts

```diff
@@ -496,4 +519,196 @@ describe('DAOFactory: ', function () {
+        [pluginSetupMockRepoAddress, 1, 1],
+        EMPTY_DATA
+      );
+
+      let DAO_INTERFACE = DAO__factory.createInterface();
+      let PSP_INTERFACE = PluginSetupProcessor__factory.createInterface();
+
+      // Prepare actions for apply Installation.
+      let applyInstallationActions = [
+        {
+          to: dao.address,
+          value: 0,
+          data: DAO_INTERFACE.encodeFunctionData('grant', [
+            dao.address,
+            psp.address,
+            ethers.utils.id('ROOT_PERMISSION'),
+          ]),
+        },
+        {
+          to: psp.address,
+          value: 0,
+          data: PSP_INTERFACE.encodeFunctionData('applyInstallation', [
+            dao.address,
+            createApplyInstallationParams(
+              plugin,
+              [pluginSetupMockRepoAddress, 1, 1],
+              permissions,
+              helpers
+            ),
+          ]),
+        },
+      ];
+
+      await expect(
+        adminPlugin.executeProposal('0x', applyInstallationActions, 0)
+      ).to.emit(psp, EVENTS.InstallationApplied);
+
+      // Prepare Update
+      const {
+        initData,
+        preparedSetupData: {
+          permissions: updatePermissions,
+          helpers: updateHelpers,
+        },
+      } = await prepareUpdate(
+        psp,
+        dao.address,
+        plugin,
+        [1, 1],
+        [1, 2],
+        pluginSetupMockRepoAddress,
+        helpers,
+        EMPTY_DATA
+      );
+
+      // Prepare actions for applyUpdate to succeed.
+      let applyUpdateActions = [
+        {
+          to: dao.address,
+          value: 0,
+          data: DAO_INTERFACE.encodeFunctionData('grant', [
+            plugin,
+            psp.address,
+            ethers.utils.id('UPGRADE_PLUGIN_PERMISSION'),
+          ]),
+        },
+        {
+          to: psp.address,
+          value: 0,
+          data: PSP_INTERFACE.encodeFunctionData('applyUpdate', [
+            dao.address,
+            createApplyUpdateParams(
+              plugin,
+              [pluginSetupMockRepoAddress, 1, 2],
+              initData,
+              updatePermissions,
+              updateHelpers
+            ),
+          ]),
+        },
+      ];
+
+      await expect(
+        adminPlugin.executeProposal('0x', applyUpdateActions, 0)
+      ).to.emit(psp, EVENTS.UpdateApplied);
+
+      // Uninstall the plugin
+      let {permissions: uninstallPermissions} = await prepareUninstallation(
+        psp,
+        dao.address,
+        plugin,
+        [pluginSetupMockRepoAddress, 1, 2],
+        updateHelpers,
+        EMPTY_DATA
+      );
+
+      // Prepare actions for apply Uninstallation.
+      let applyUninstallationActions = [
+        {
+          to: psp.address,
+          value: 0,
+          data: PSP_INTERFACE.encodeFunctionData('applyUninstallation', [
+            dao.address,
+            createApplyUninstallationParams(
+              plugin,
+              [pluginSetupMockRepoAddress, 1, 2],
```

Comment:
Why is the version number required to uninstall a plugin? Isn't the plugin address and the pluginRepo's address (to resolve the setup) enough?